corva/models/stream.py

```diff
@@ -1,26 +1,43 @@
 from __future__ import annotations
 
-from typing import Dict, List, Optional, Union
+from typing import Dict, List, Literal, Optional, Union
 
 import pydantic
 
 from corva.models.base import BaseContext, BaseEvent, CorvaBaseModel
 
 
-class RecordData(CorvaBaseModel):
-    hole_depth: Optional[float] = None
-    weight_on_bit: Optional[int] = None
-    state: Optional[str] = None
-
-
 class Record(CorvaBaseModel):
-    timestamp: Optional[int] = None
     asset_id: int
-    company_id: int
-    version: int
-    measured_depth: Optional[float] = None
-    collection: str
-    data: RecordData
+
+    data: Optional[dict] = None
+    metadata: Optional[dict] = None
+
+    timestamp: Optional[int] = pydantic.Field(
+        None,
+        description='Timestamp (Unix epoch time). Only present for time based streams',
+    )
+    measured_depth: Optional[float] = pydantic.Field(
+        None,
+        description='Measured depth (ft). Only present for depth based streams',
+    )
+
+    version: Optional[int] = pydantic.Field(None, description='Version of the record')
+    app: Optional[str] = pydantic.Field(
+        None, description='App which generated the data'
+    )
+    company_id: Optional[int] = None
+    provider: Optional[str] = None
+    collection: Optional[str] = None
+
+    @pydantic.root_validator(pre=True)
+    def require_timestamp_or_measured_depth(cls, values):
+        if 'timestamp' in values and 'measured_depth' not in values:
```

Comment:
```suggestion
        if 'timestamp' not in values and 'measured_depth' not in values:
            raise ValueError('Either timestamp or measured_depth is required')
            
        return values
```

Reply:
I thought that there should always be either timestamp or measured_depth?

Reply:
That's correct, that's what this logic checks.

Reply:
simplified the logic to one if